### PR TITLE
Create logout test

### PR DESCRIPTION
**Pull request by Dmaza86**

### CODE REVIEW

File: src/test/kotlin/pageobjects/LogoutPage.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import org.fluentlenium.core.annotation.PageUrl
+import org.fluentlenium.core.domain.FluentWebElement
+import org.openqa.selenium.support.FindBy
+import pageobject.AbstractScreen
+
+@PageUrl("https://admin.goqr.co/sign-in")
+class LogoutPage : AbstractScreen() {
+
+
```
Comment (Jose-Luis-Nunez) [inline src/test/kotlin/pageobjects/LogoutPage.kt:11]: Delete line

File: src/test/kotlin/pageobjects/LogoutPage.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    @FindBy(css = ".dropdown")
+    private lateinit var menu: FluentWebElement
+
+    @FindBy(css = "div.dropdown-menu a:last-child")
+    private lateinit var logoutButton: FluentWebElement
+
+    fun logout() {
+        clickOnElement(menu)
```
Comment (Jose-Luis-Nunez) [inline src/test/kotlin/pageobjects/LogoutPage.kt:19]: I would extract the click on menu in an extra function

File: src/test/kotlin/tests/goqr/logout/LogoutTest.kt [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import org.junit.jupiter.api.Test
+import pageobjects.LogoutPage
+import pageobjects.SignInPage
+import tests.SeleniumTest
+
+@Browser(dimension = Breakpoint.XXLARGE)
+class LogoutTest : SeleniumTest() {
+
+    @Page
+    lateinit var signInPage: SignInPage
```
Comment (Jose-Luis-Nunez) [inline src/test/kotlin/tests/goqr/logout/LogoutTest.kt:17]: New line

File: src/test/kotlin/tests/goqr/logout/LogoutTest.kt [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            fillSignInForm(email, password)
+        }
+    }
+
+    @Test
+    fun `user can logout`() {
+      logoutPage{
+          logout()
+      }
+        assertThat(signInPage.url).isEqualTo(url())
```
Comment (Jose-Luis-Nunez) [inline src/test/kotlin/tests/goqr/logout/LogoutTest.kt:37]: Reformat code

Comment (Jose-Luis-Nunez) [inline src/test/kotlin/tests/goqr/logout/LogoutTest.kt:37]: I would not assert on an url better on an element that is only visible when the user is logout

File: src/test/kotlin/tests/goqr/logout/LogoutTest.kt [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+@Browser(dimension = Breakpoint.XXLARGE)
+class LogoutTest : SeleniumTest() {
+
+    @Page
+    lateinit var signInPage: SignInPage
+    @Page
+    lateinit var logoutPage: LogoutPage
+
+    @BeforeEach
```
Comment (Jose-Luis-Nunez) [inline src/test/kotlin/tests/goqr/logout/LogoutTest.kt:21]: The idea is nice but I would just put all in the test function

File: src/test/kotlin/pageobjects/LogoutPage.kt [hunk 6]
```diff
@@ -0,0 +1,22 @@
+package pageobjects
+
+import org.fluentlenium.core.annotation.PageUrl
+import org.fluentlenium.core.domain.FluentWebElement
+import org.openqa.selenium.support.FindBy
+import pageobject.AbstractScreen
+
+@PageUrl("https://admin.goqr.co/sign-in")
+class LogoutPage : AbstractScreen() {
```
Comment (Jose-Luis-Nunez) [inline src/test/kotlin/pageobjects/LogoutPage.kt:9]: I think this is not the logout page right ? I think this a screen where you are logged in right ? Or is it maybe the menu element ?

**Review by Jose-Luis-Nunez — CHANGES_REQUESTED** [review]

File: src/test/kotlin/pageobjects/LogoutPage.kt (hunk 6 above)
Comment (Jose-Luis-Nunez) [reply src/test/kotlin/pageobjects/LogoutPage.kt:9]: yeah better use the LanguageMenuPage and rename to only menu and put here the logout stuff

**Comment by Jose-Luis-Nunez** [comment]
you can have a package menu and this could have 2 subfolders logout and translation

File: src/test/kotlin/pageobjects/LogoutPage.kt (hunk 1 above)
Comment (Dmaza86) [reply src/test/kotlin/pageobjects/LogoutPage.kt]: done

File: src/test/kotlin/pageobjects/LogoutPage.kt (hunk 2 above)
Comment (Dmaza86) [reply src/test/kotlin/pageobjects/LogoutPage.kt]: done

File: src/test/kotlin/tests/goqr/logout/LogoutTest.kt (hunk 3 above)
Comment (Dmaza86) [reply src/test/kotlin/tests/goqr/logout/LogoutTest.kt:16]: done

File: src/test/kotlin/tests/goqr/logout/LogoutTest.kt (hunk 4 above)
Comment (Dmaza86) [reply src/test/kotlin/tests/goqr/logout/LogoutTest.kt]: done

Comment (Dmaza86) [reply src/test/kotlin/tests/goqr/logout/LogoutTest.kt]: done

File: src/test/kotlin/tests/goqr/logout/LogoutTest.kt (hunk 5 above)
Comment (Dmaza86) [reply src/test/kotlin/tests/goqr/logout/LogoutTest.kt]: done

File: src/test/kotlin/pageobjects/LogoutPage.kt (hunk 6 above)
Comment (Dmaza86) [reply src/test/kotlin/pageobjects/LogoutPage.kt]: done

**Comment by Dmaza86** [comment]
> you can have a package menu and this could have 2 subfolders logout and translation

**Review by Jose-Luis-Nunez — CHANGES_REQUESTED** [review]

File: src/test/kotlin/pageobjects/SignInPage.kt [hunk 7]
```diff
@@ -22,4 +22,8 @@ class SignInPage : AbstractScreen() {
         passwordInput.type(password)
         clickOnElement(singInButton)
     }
+
+    fun getSignInButton():FluentWebElement{
```
[the remark this answers is not in the document]
Comment (Dmaza86) [reply src/test/kotlin/pageobjects/SignInPage.kt:26]: done

File: src/test/kotlin/tests/goqr/logout/LogoutTest.kt [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+import tests.SeleniumTest
+
+@Browser(dimension = Breakpoint.XXLARGE)
+class LogoutTest : SeleniumTest() {
+
+    @Page
+    lateinit var signInPage: SignInPage
+
+    @Page
+    lateinit var logoutPage: LogoutPage
```
[the remark this answers is not in the document]
Comment (Dmaza86) [reply src/test/kotlin/tests/goqr/logout/LogoutTest.kt]: done

File: src/test/kotlin/tests/goqr/logout/LogoutTest.kt [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+import tests.SeleniumTest
+
+@Browser(dimension = Breakpoint.XXLARGE)
+class LogoutTest : SeleniumTest() {
+
+    @Page
+    lateinit var signInPage: SignInPage
+
+    @Page
+    lateinit var menuPage: MenuPage
```
[the remark this answers is not in the document]
Comment (Dmaza86) [reply src/test/kotlin/tests/goqr/logout/LogoutTest.kt]: done

File: src/test/kotlin/pageobjects/MenuPage.kt [hunk 10]
```diff
@@ -6,10 +6,10 @@ import org.openqa.selenium.support.FindBy
 import pageobject.AbstractScreen
 
 @PageUrl("https://admin.goqr.co/sign-in")
-class LanguageMenuPage : AbstractScreen() {
+class MenuPage : AbstractScreen() {
```
[the remark this answers is not in the document]
Comment (Dmaza86) [reply src/test/kotlin/pageobjects/MenuPage.kt]: done

File: src/test/kotlin/pageobjects/MenuBar.kt [hunk 11]
```diff
@@ -23,8 +23,22 @@ class LanguageMenuPage : AbstractScreen() {
...
+    private lateinit var downloadQrButton: FluentWebElement
+
+     fun accessToLanguageMenu() {
+        clickOnElement(menu)
+    }
+
+    fun accessToMenu() {
+        waitUntilElementDisplayed(downloadQrButton)
+        clickOnElement(menu)
+    }
```
Comment (Jose-Luis-Nunez) [inline src/test/kotlin/pageobjects/MenuBar.kt:39]: we need still to improve this

**Review by Jose-Luis-Nunez — CHANGES_REQUESTED** [review]

File: src/test/kotlin/pageobjects/MenuBar.kt (hunk 11 above)
Comment (Dmaza86) [reply src/test/kotlin/pageobjects/MenuBar.kt]: done

**Review by Jose-Luis-Nunez — APPROVED** [review]

**Pull request merged**